realm/realm-library/src/main/java/io/realm/internal/FieldDescriptor.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2016 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.realm.internal;
+
+import java.util.Arrays;
+
+import io.realm.RealmFieldType;
+
+public class FieldDescriptor {
+
+    private long[] columnIndices;
+    private RealmFieldType fieldType;
+    private String fieldName;
+    private boolean searchIndex;
+
+    public FieldDescriptor(Table table, String fieldDescription, boolean allowLink, boolean allowList) {
+        if (fieldDescription == null || fieldDescription.isEmpty()) {
+            throw new IllegalArgumentException("Non-empty field name must be provided");
+        }
+        if (fieldDescription.startsWith(".") || fieldDescription.endsWith(".")) {
+            throw new IllegalArgumentException("Illegal field name. It cannot start or end with a '.': " + fieldDescription);
+        }
+        if (fieldDescription.contains(".")) {
+            // Resolve field description down to last field name
+            String[] names = fieldDescription.split("\\.");
+            long[] columnIndices = new long[names.length];
+            for (int i = 0; i < names.length - 1; i++) {
+                long index = table.getColumnIndex(names[i]);
+                if (index < 0) {
```

Comment:
As long as this can only return `Table.NO_MATCH` and not `Table.NO_PRIMARY_KEY` which is `-2`, while `NO_MATCH` is `-1`